src/org/thoughtcrime/securesms/ConversationFragment.java

```diff
@@ -828,6 +828,11 @@ public void onShowFullClicked(DcMsg messageRecord) {
         startActivity(intent);
         getActivity().overridePendingTransition(R.anim.slide_from_right, R.anim.fade_scale_out);
       }
+
+      @Override
+      public void onDownloadClicked(DcMsg messageRecord) {
+        dcContext.downloadFullMsg(messageRecord.getId());
```

Comment:
if user tap the download button several times, will core ignore calls to `downloadFullMsg` if the download is already ongoing? or should `onDownloadClicked` ignore calls while it is already downloading? 
(offtopic: I have issues with this buttons I am never sure if tapping them works because they are styled like links without any state change on taps, so if the download failed quickly I am not sure if it failed or I missed the button)

Reply:
multiple calls to downloadFullMsg() should be ingored by core.

once the download is in progress, the ui showns the usual tiny loading animation, however, it would be nice to also disable the download button to reflect that state.

at some point, a "cancel download" appearing instead would maybe be nice, however, we just do not have that functionality core-side currently - and it is also not that easy iirc.

Reply:
i tweaked the button a bit, it gets disabled now and shows "Downloading..." at that time. would be nice to have that as grey, however, that needs tweaking of some xml; i did not dived into that.

Reply:
it is much better now +1